### ISSUE TITLE
ChemPropFeaturizer refits target scaler per split instead of fit-on-train, apply-to-val/test

### ISSUE THREAD

**Issue by smcolby** [OpenADMET/openadmet_models#579]
## Summary

`ChemPropFeaturizer.featurize` refits a fresh `StandardScaler` on every call instead of fitting on the training split and reusing that scaler for validation and test. Target normalization should be fit on train only and applied elsewhere.

## Where

In `openadmet/models/features/chemprop.py`, `ChemPropFeaturizer.featurize` normalizes targets with:

```python
if self.normalize_targets:
    scaler = dataset.normalize_targets()   # no scaler argument -> refits on this call's targets
```

This runs independently for each split. In `openadmet/models/anvil/workflow.py` the deep-learning training path calls:

```python
train_dataloader, _, train_scaler, train_dataset = self.feat.featurize(X_train, y_train, train=True)
val_dataloader,   _, _,            val_dataset   = self.feat.featurize(X_val,   y_val)            # refits a val scaler
test_dataloader,  _, _,            test_dataset  = self.feat.featurize(X_test,  y_test)           # refits a test scaler
```

Only `train_scaler` is kept (passed to `model.build(scaler=...)` for the output transform). The val and test scalers are discarded, but each call has already normalized that split's targets in place using its own mean/std.

## Effect

During validation, chemprop keeps `predictor.output_transform` in train mode (`on_validation_model_eval`), so predictions stay in **train**-normalized space while validation targets are in **val**-normalized space. `val_loss` and the per-task validation metrics therefore compare quantities in two slightly different scales, offset by the train-vs-val mean/std difference. The discrepancy is small for random splits and larger for cluster/scaffold splits where the train and val target distributions genuinely differ.

## What is not affected

- **Not data leakage.** The val/test scalers are fit only on their own split; no val/test information flows into training, and model weights are fit on the training split alone.
- **Trained model artifact is unaffected.** Validation is used only for monitoring and early stopping; the output transform uses the train scaler, so inference correctly returns raw-unit predictions.
- **Post-hoc test metrics are unaffected** when computed from raw `y_test` against raw predictions (`model.predict`), since that path does not use the refit test-target normalization.

The practical impact is confined to in-training `val_loss` / val metrics, which slightly distorts the early-stopping signal (mostly benign because the distortion is roughly constant across epochs, so the selected epoch is largely preserved).

## Suggested fix

Give `ChemPropFeaturizer.featurize` an optional `scaler` parameter; when provided, call `dataset.normalize_targets(scaler)` to apply it rather than refit. Thread the `train_scaler` returned from the training featurization into the val and test featurize calls in `workflow.py` (both the single-model and ensemble paths).

## Priority

Low. Tracking for a later fix. It does not invalidate cross-run comparisons that share the same pipeline, and it does not affect the final raw-space evaluation metrics.
